test/puppetlabs/puppetdb/catalogs_test.clj

```diff
@@ -176,3 +179,46 @@
         (is (not (contains? wire-catalog :metadata)))
         (is (= (dissoc catalog :api_version)
                wire-catalog))))))
+
+(deftest test-v6-conversion
+  (testing "v5->v6"
+    (let [v5-catalog (get-in wire-catalogs [5 :empty])]
```

Comment:
I wanted to add some schema tests here (like I did for reports) but as I added that, I stubbed my toe on some old v4/v5 code that didn't get cleaned up properly and the change was getting bigger. I wrote this up as [PDB-1630](https://tickets.puppetlabs.com/browse/PDB-1630) to be tackled later.